### ISSUE TITLE
Running the trained model from c++

### ISSUE THREAD

**Issue by arnonkahani** [Russell91/ReInspect#28]
Hello,
I'm quite new at Caffe and I'm trying to achieve a object detection system in C++
Is it possible to run the a trained model produced by ReInspect in C++?
And if so can you please point out some guide lines.
Thanks

**Comment by Russell91** [comment]
Yes, absolutely. If you look at apollocaffe, there is a c++ api for running the network inside the apollonet.hpp file. You should be able to initialize the net and call forward from there. You'll have to save the proto from python and then run those same protos in the forward pass. I'm not actively developing apollocaffe anymore, so you're mostly on your own. But the apollonet code is actually pretty clean if you've got some c++ skills.

**Comment by arnonkahani** [comment]
Thanks very much for the quick response!
One last thing , can you please give a quick explanation about the deploy flag in the forward function.
Thanks again.

**Comment by Russell91** [comment]
There are some layers, such as dropout, which behave differently at test
time than during training. The deploy flag is the same as in the core caffe
branch, and e.g. in the case of dropout it tells the network not to
randomly drop out neurons.
